### ISSUE TITLE
add api to for distro packages maintainer

### ISSUE THREAD

**Issue by glensc** [fedora-infra/anitya#42]
as distribution maintainer, i would like to do over api:
- create project<>distribution package mapping (project_id (numeric), distribution name, distribution package name) 
- add project (name, homepage, backend, backend specific options)
- rename package mapping for distribution (project_id, distribution name, old name(?), new name)

**Comment by ralphbean** [comment]
Agreed.  This could be expanded.  We would want all of those new API endpoints to require authentication.

Noting here that here's an example of using parts of the existing anitya API (unauthenticated):  https://github.com/ralphbean/lightsaber/blob/develop/roles/development/files/bin/check-anitya.py

**Comment by ralphbean** [comment]
And here's an example of doing some things authenticated:  https://github.com/fedora-infra/the-new-hotness/blob/develop/hotness/anitya.py

**Comment by jeremycline** [comment]
With the v2 API, the only feature not yet supported is 

* rename package mapping for distribution (project_id, distribution name, old name(?), new name).

We need to add PUT request support to the packages endpoint
